### DISCUSSION TITLE
Opus.dll vs opusenc.exe

### DISCUSSION THREAD

**Discussion by szabiz** [enzo1982/freac#483] [Help & Support]
First of all, I would like to thank you for getting the opus 1.4 version into development so quickly.

My question is, what is the difference between Opus.dll and opusenc.exe, because it is true that both identify themselves as opus 1.4, but the sound is not the same, opus.dll is better for some files, especially mono sound. why do some programs use this, and others that. Thank you in advance for your answer.

szabiz

**Comment by enzo1982** [comment]
There should be no difference in audio quality between Opus.dll and opusenc.exe when using the same compression settings.

The difference could potentially come from different resamplers used by fre:ac and opusenc. Opus needs audio data sampled at 48 kHz, so resampling is needed when the input files have a different sample rate. fre:ac uses the high quality libsamplerate resampler to do this. Other programs may rely on the integrated resampler of opusenc or use a different resampler of their own.

That's just one possibility, though. The actual reason why you're getting inferior quality with opusenc may be different. I'd always recommend to double check the compression settings in both programs first.

**Comment by szabiz** [comment]
Yes, you may be right, because I tried different applications to convert .wav 44100 khz mono to .opus 48000 kHz. 32 bit vb, mono.

At the same time, it is interesting that freac produced the best results. I experimented with the following programs: Aimp audio converter, XmediaRecode, TAudioCoverter, FFMPEG Audio Encoder, LameXp, Audacity, foobar2000, WavePad Sound Editor.

What all programs had in common, including freac, was that the speech voice was sharpened and distorted a very little. I don't understand why the opus generates higher frequencies above the existing ones when I put it on full band.

**Reply by enzo1982** [reply]
It's not so much that Opus generates those frequencies. Those are just side-effects of encoding, especially at low bitrates. At higher bitrates, the codec has more bits available to suppress such side-effects/artifacts.

**Reply by szabiz** [reply]
I understand, thank you very much for the answer.
